src/actions/languages/python/motion.ts

```diff
@@ -0,0 +1,232 @@
+import { RegisterAction } from '../../base';
+import { VimState } from '../../../state/vimState';
+import { BaseMovement } from '../../baseMotion';
+import { Position, TextDocument } from 'vscode';
+
+interface LineInfo {
+  line: number;
+  indentation: number;
+  text: string;
+}
+
+interface StructureElement {
+  type: 'function' | 'class';
+  start: Position;
+  end: Position;
+}
+
+/*
+ * Utility class used to parse the lines in the document and
+ * determine class and function boundaries
+ *
+ * The class keeps track of two positions: the ORIGINAL and the CURRENT
+ * using their relative locations to make decisions.
+ */
+export class PythonDocument {
+  _document: TextDocument;
+  structure: StructureElement[];
+
+  static readonly reOnlyWhitespace = /\S/;
+  static readonly reLastNonWhiteSpaceCharacter = /(?<=\S)\s*$/;
+  static readonly reDefOrClass = /^\s*(def|class) /;
+  static readonly reDef = /def/;
+
+  constructor(document: TextDocument) {
+    this._document = document;
+    const parsed = PythonDocument._parseLines(document);
+    this.structure = PythonDocument._parseStructure(parsed);
+  }
+
+  /*
+   * Generator of the lines of text in the document
+   */
+  static *lines(document: TextDocument): Generator<string> {
+    for (let index = 0; index < document.lineCount; index++) {
+      yield document.lineAt(index).text;
+    }
+  }
+
+  /*
+   * Calculate the indentation of a line of text.
+   * Lines consisting entirely of whitespace of "starting" with a comment are defined
+   * to have an indentation of "null".
+   */
+  static _indentation(line: string): number | null {
```

Comment:
`undefined` -> `null` here and in `_parseLine`. Sorry to nitpick about this 🙂

Reply:
Done.